### ISSUE TITLE
llvm-objdump does not understand DT_MIPS_XHASH in mips binaries

### ISSUE THREAD

**Issue by kraj** [llvm/llvm-project#53996]
when dumping a binary build with gnu-hash for mips, it emits DT_MIPS_XHASH into object but such a binary when dumped with llvm-objdump says

```
  <unknown:>0x70000036 0x00000168
```
where as GNU objdump does understand it
```
  DT_MIPS_XHASH        0x00000168
```

Attached is an example use `llvm-objump -p` to dump it.

[libx264.tar.gz](https://github.com/llvm/llvm-project/files/8119853/libx264.tar.gz)

**Comment by MaskRay** [comment]
I think `DT_MIPS_XHASH` is mis-designed: https://maskray.me/blog/2021-08-29-all-about-global-offset-table#dt_mips_local_gotno-and-dt_mips_symtabno-dt_mips_gotsym

It's not binutils's default so perhaps the simplest approach is to not emit `DT_MIPS_XHASH` (examine `--hash-style`)

**Comment by jh7370** [comment]
> I think `DT_MIPS_XHASH` is mis-designed: https://maskray.me/blog/2021-08-29-all-about-global-offset-table#dt_mips_local_gotno-and-dt_mips_symtabno-dt_mips_gotsym
> 
> It's not binutils's default so perhaps the simplest approach is to not emit `DT_MIPS_XHASH` (examine `--hash-style`)

As I understand it, the only thing we need to do here is interpret it in the dynamic table output right? It should be trivially easy to do. If there are tools currently emitting the tag, we should do this,

**Comment by kraj** [comment]
Thanks can this be backported to 14.x release branch too

**Comment by tstellar** [comment]
@atanasyan Is this OK to backport? https://github.com/llvm/llvm-project/commit/3c840e3c00e910c47a3f61f755fdc402d51e9fb6

**Comment by atanasyan** [comment]
> @atanasyan Is this OK to backport? [3c840e3](https://github.com/llvm/llvm-project/commit/3c840e3c00e910c47a3f61f755fdc402d51e9fb6)

@tstellar  Yes, it can be backported.

**Comment by tstellar** [comment]
Merged: 4c9110a5f37b
